packages/picasso/src/RichTextEditor/RichTextEditor.tsx

```diff
@@ -0,0 +1,230 @@
+  autoFocus?: boolean
+  /** Default value in [HAST](https://github.com/syntax-tree/hast) format */
+  defaultValue?: ASTType
+  /**
+   * This Boolean attribute indicates that the user cannot interact with the control.
+   */
+  disabled?: boolean
+  /** unique identifier */
+  id: string
+  /**
+   * Indicate wether the editor is in an error state
+   */
+  error?: boolean
+  /**
+   * The maximum number of characters that the user can enter.
+   * If this value isn't specified, the user can enter an unlimited
+   * number of characters.
+   */
+  maxLength?: number
+  /**
+   * The minimum number of characters required that the user should enter.
+   */
+  minLength?: number
+  /**
+   * Custom counter message for minLength
+   */
+  minLengthMessage?: CounterMessageSetter
+  /**
+   * Custom counter message for maxLength
+   */
+  maxLengthMessage?: CounterMessageSetter
+  /**
+   * Callback on text change
+   */
+  onChange?: (value: string) => void
+  /**
+   * Callback for blur event
+   */
+  onBlur?: () => void
+  /**
+   * Callback for focus event
+   */
+  onFocus?: () => void
+  /** The placeholder attribute specifies a short hint that describes the expected value of a text editor. */
+  placeholder?: string
+  testIds?: {
+    wrapper?: string
+    editor?: string
+    headerSelect?: string
+    boldButton?: string
+    italicButton?: string
+    unorderedListButton?: string
+    orderedListButton?: string
+  }
+}
+
+const useStyles = makeStyles<Theme>(styles, {
+  name: 'RichTextEditor'
+})
+
+export const RichTextEditor = forwardRef<HTMLDivElement, Props>(
+  function RichTextEditor(
+    {
+      'data-testid': dataTestId,
+      autoFocus = false,
+      className,
+      defaultValue,
+      disabled,
+      id,
+      onChange = noop,
+      onFocus = noop,
+      onBlur = noop,
+      placeholder,
+      minLength,
+      maxLength,
+      minLengthMessage,
+      maxLengthMessage,
+      style,
+      error,
+      testIds
+    },
+    ref
+  ) {
+    const classes = useStyles()
+    const toolbarRef = useRef<HTMLDivElement | null>(null)
+    const editorRef = useRef<HTMLDivElement | null>(null)
+    const wrapperRef = useRef<HTMLDivElement | null>(null)
+    const { dispatch, state } = useTextEditorState()
+
+    const { handleSelectionChange } = useOnSelectionChange({ dispatch })
+    const { handleTextFormat } = useOnTextFormat({ dispatch })
+    const {
+      handleBold,
+      handleItalic,
+      handleHeader,
+      handleOrdered,
+      handleUnordered
+    } = useToolbarHandlers({
+      editorRef,
+      handleTextFormat,
+      format: state.toolbar.format
+    })
+
+    const { isEditorFocused, handleFocus, handleBlur } = useOnFocus({
+      autoFocus,
+      editorRef,
+      toolbarRef,
+      wrapperRef,
+      onFocus,
+      onBlur,
+      dispatch
+    })
+
+    const [defaultValueInHtml] = useState(
+      defaultValue ? hastUtilToHtml(hastSanitize(defaultValue)) : defaultValue
```

Comment:
Because we are not wrapping the default in a function `hastUtilToHtml` is being run every re-render. React already supports the first argument as a function and only run it on creation
```suggestion
      () => defaultValue ? hastUtilToHtml(hastSanitize(defaultValue)) : defaultValue
```

Reply:
nice, didn't know it.